pallets/rc-migrator/src/accounts.rs

```diff
@@ -388,19 +374,56 @@ impl<T: Config> AccountsMigrator<T> {
 			}
 		}
 
+		let ed = <T as Config>::Currency::minimum_balance();
 		let holds: Vec<IdAmount<T::RuntimeHoldReason, T::Balance>> =
 			pallet_balances::Holds::<T>::get(&who).into();
 
 		for hold in &holds {
-			if let Err(e) =
-				<T as Config>::Currency::release(&hold.id, &who, hold.amount, Precision::Exact)
-			{
+			let IdAmount { id, amount } = hold.clone();
+			let free = <T as Config>::Currency::balance(&who);
+
+			// When the free balance is below the minimum balance and we attempt to release a hold,
+			// the `fungible` implementation would burn the entire free balance while zeroing the
+			// hold. To prevent this, we partially release the hold just enough to raise the free
+			// balance to the minimum balance, while maintaining some balance on hold. This approach
+			// prevents the free balance from being burned.
+			// This scenario causes a panic in the test environment - see:
+			// https://github.com/paritytech/polkadot-sdk/blob/35e6befc5dd61deb154ff0eb7c180a038e626d66/substrate/frame/balances/src/impl_fungible.rs#L285
```

Comment:
We probably need to give the whole account migration another throughout review at the end, so am just going to rely on tests for now and merge.